### PR TITLE
Rewrite tests

### PR DESCRIPTION
**Pull request by alicebartlett**
This PR rewrites all the tests in o-date.
I've tried to improve:

- the readability of the tests
- the modularity of the tests (each test should test that one function does one thing)

Feedback/comments welcome @rowanmanning

### CODE REVIEW

**Comment by rowanmanning** [comment]
The spacing across the test file is a bit wonk, should all be tabs

File: test/date.test.js [hunk 1]
```diff
@@ -1,150 +1,375 @@
...
 
-		expect(oDate.format(date)).to.be(expected);
-		expect(oDate.format(isoDate)).to.be(expected);
-		expect(oDate.format(date, 'datetime')).to.be(expected);
-		expect(oDate.format(date, 'date')).to.be(expected.replace(' 6:37 am', ''));
-		// testing padding
-		expect(oDate.format(new Date(2000, 5, 15, 6, 7), 'datetime')).to.be('June 15, 2000 6:07 am');
-	});
+		it('returns a date object if the date passed in is valid', function() {
+			const valid_dates = [
```
Comment (rowanmanning) [inline test/date.test.js]: Should be `validDates`.

File: test/date.test.js [hunk 2]
```diff
@@ -1,150 +1,375 @@
...
-
-		expect(oDate.format(date, 'yyyy yy')).to.be('2000 00');
-		expect(oDate.format(date, 'MMMM MMM MM M')).to.be('February Feb 02 2');
-		expect(oDate.format(date, 'EEEE EEE')).to.be('Saturday Sat');
-		expect(oDate.format(date, 'd dd')).to.be('5 05');
-		expect(oDate.format(date, 'h hh')).to.be('6 06');
-		expect(oDate.format(date, 'm mm')).to.be('7 07');
-		expect(oDate.format(date, 'a')).to.be('am');
-		expect(oDate.format(date, 'This is \\a co\\mmon string mm')).to.be('This is a common string 07');
+      for (let i = 0; i < valid_dates.length; i++) {
```
Comment (rowanmanning) [inline test/date.test.js]: This could be a `for of` loop, would simplify slightly

File: test/date.test.js [hunk 3]
```diff
@@ -1,150 +1,375 @@
...
+      spyOn(oDate, 'isToday').and.returnValue(false);
+      spyOn(oDate, 'isYesterday').and.returnValue(false);
+      spyOn(oDate, 'format').and.returnValue('a pretend date string');
+
+      expect(oDate.ftTime(someDate)).toBe('a pretend date string');
+    });
+  });
+
+  describe('isNearFuture', function() {
+    it('returns true if interval less than 5 minutes, false otherwise', function(){
```
Comment (rowanmanning) [inline test/date.test.js]: Probs a bit picky, but this could be split into two separate tests

**Comment by rowanmanning** [comment]
In tests for other components (e.g. o-header) we seem to be moving towards using arrow functions for all of the `describe`/`it` blocks. I think it's worth converting all the regular functions in these tests, and is safe to do so.

**Comment by alicebartlett** [comment]
Cool  thanks @rowanmanning! I've addressed your comments.

**Comment by rowanmanning** [comment]
Looks great 👍 feel free to merge

**Pull request merged**